Sources/MapboxMaps/Foundation/Extensions/Encodable+ToJSON.swift

```diff
@@ -16,4 +16,13 @@ internal extension Encodable {
         // on `first` being non-nil
         return array.first!
     }
+
+    /// Given an Encodable object return the JSON representation as a string
+    /// - Throws: Errors occurring during conversion.
+    /// - Returns: A string with JSON representing the object.
+    func toString() throws -> String {
+        let data = try JSONEncoder().encode(self)
+
+        return String(decoding: data, as: UTF8.self)
```

Comment:
What would happen if the data is not UTF-8 representable? Method does not throw and do not return optional string. Would it be a runtime NSExpection then?
Would it be better to use safe APIs instead?

Reply:
Couldn't find the answer to your question - converted it to a failable initializer :)